src/dev/src/app/datagrid/utils/date-interval-filter.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component, EventEmitter } from '@angular/core';
+import { User } from '../inventory/user';
+import { SerializableFilter } from '../../../../../clr-angular/data/datagrid/interfaces/serializable.filter.interface';
+import { DateIntervalFilterStateInterface } from '../../../../../clr-angular/data/datagrid/interfaces/date.interval.filter.state.interface';
+import { from } from 'rxjs/internal/observable/from';
+
+@Component({
+  selector: 'clr-datagrid-date-interval-filter-demo',
+  template: `
+    <div (click)="setFilters()">
+      <span (click)="setToDate()">
+        From: <input *ngIf="visible" type="text" [(clrDate)]="from">
+      </span>
+      To: <input *ngIf="visible" type="text" [(clrDate)]="to">
+      <clr-icon role="button" shape="trash" (click)="resetFilter()"></clr-icon>
+    </div>`,
+  styleUrls: ['../datagrid.demo.scss'],
+})
+export class DateIntervalFilter implements SerializableFilter<User> {
+  from: Date;
+  to: Date;
+  visible = true;
+  changes: EventEmitter<any> = new EventEmitter<any>(false);
+
+  constructor() {
+    this._id = Math.random().toString();
+  }
+
+  private _id: string;
+
+  get id() {
+    return this._id;
+  }
+
+  public get filterState(): DateIntervalFilterStateInterface {
+    return {
+      id: this.id,
+      type: 'DateIntervalFilter',
+      from: this.from,
+      to: this.to,
+    };
+  }
+
+  public set filterState(state: DateIntervalFilterStateInterface) {
+    this.from = state.from;
+    this.to = state.to;
+  }
+
+  setId(id: string): void {
```

Comment:
I would use a property and not a set-method, since it doesn't do anything anyway. Less code is better and properties are easy.